File: four/delia/maxcounter.js
function solution(N, A) {
    counters = [];
    for(i=0;i<N;i++) counters[i] = 0;  
    maxCounter = 0;
    maxBernardy = 0; 
    for(j=0;j<A.length;j++){
        if(A[j]==N+1) maxBernardy = maxCounter; 
        else if (A[j] < N+1) { 
            counters[A[j]-1] = 1 + Math.max(counters[A[j]-1],maxBernardy);
            maxCounter = Math.max(maxCounter, counters[A[j]-1]);
        }
    }
    for(k=0;k<N;k++) counters[k] = Math.max(maxBernardy,counters[k]);
    return counters;
}

File: four/francis/maxcounter.py
def solution(N, A):
    max_value = 0
    tmp_max_value = 0
    counters = {}
    for i in A:
        if 1 <= i <= N:
            if i in counters:
                counters[i] += 1
            else:
                counters[i] = 1
            tmp_max_value = max(tmp_max_value, counters[i])
        else:
            max_value += tmp_max_value
            tmp_max_value = 0
            counters = {}
    return [max_value + (counters[x] if x in counters else 0) for x in range(1, N+1)]
